File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcProfileDef/IfcParameterizedProfileDef/IfcLShapeProfileDef/IfcLShapeProfileDef.vb
Public Class IfcLShapeProfileDef Inherits IfcParameterizedProfileDef
    Public Depth As IfcPositiveLengthMeasure
    Public Width As IfcPositiveLengthMeasure
    Public Thickness As IfcPositiveLengthMeasure
    Public FilletRadius As IfcNonNegativeLengthMeasure
    Public EdgeRadius As IfcNonNegativeLengthMeasure
    Public LegSlope As IfcPlaneAngleMeasure

    ' === WHERE CLAUSES ===
    ' ValidThickness : (Thickness < Depth) AND (NOT(EXISTS(Width)) OR (Thickness < Width))
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcProfileDef/IfcParameterizedProfileDef/IfcLShapeProfileDef/IfcLShapeProfileDef.java
public class IfcLShapeProfileDef extends IfcParameterizedProfileDef {
    public IfcPositiveLengthMeasure Depth;
    public IfcPositiveLengthMeasure Width;
    public IfcPositiveLengthMeasure Thickness;
    public IfcNonNegativeLengthMeasure FilletRadius;
    public IfcNonNegativeLengthMeasure EdgeRadius;
    public IfcPlaneAngleMeasure LegSlope;

    // === WHERE CLAUSES ===
    // ValidThickness : (Thickness < Depth) AND (NOT(EXISTS(Width)) OR (Thickness < Width))
}
